doc/telemetry.md

```diff
@@ -50,6 +50,10 @@ def add(x, y):
 add(1, 41)
 ```
 
+```{note}
+Event names are normalized by replacing underscores (`_`) with hyphens (`-`).
```

Comment:
I think we can clarify and add some examples of how the event name should be formatted.
Later in the tests, we have `somepackage-do_stuff-success` and `somepackage-SomeObject-do_stuff-started` but I'm not sure we actually follow these rules in our codebase. Maybe add it to contributing repo?

Reply:
I think we want people not to add custom names, but just add the `log_call` decorator, only in exceptional cases, they should add a custom name. this way we'll control the formatting